### PR TITLE
feat: add disable overlay functionality; default to false

### PR DESCRIPTION
**Pull request by ankitu**
This commit adds disable freeze overlay functionality, usage `disableOverlay` option can be passed to the preboot initial config as follows:
```
PrebootModule.withConfig({ appRoot: 'app-root', disableOverlay: true })
```
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/preboot/blob/master/CONTRIBUTING.md#commit-message-format
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What modules are related to this pull-request**
- [ ] server side
- [ ] client side
- [ ] inline
- [ ] build process
- [ ] docs
- [ ] tests

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
remove unused proxy imports from some test files

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: src/lib/api/event.recorder.ts [hunk 1]
```diff
@@ -262,8 +265,8 @@ export function createListenHandler(
       prebootData.activeNode = undefined;
     }
 
-    // if we are freezing the UI
-    if (eventSelector.freeze) {
+    // if overlay is not disabled and we are freezing the UI
+    if (!(opts.disableOverlay || false) && eventSelector.freeze) {
```
Comment (mgol) [inline src/lib/api/event.recorder.ts]: `|| false`?

File: src/lib/api/event.recorder.ts [hunk 2]
```diff
@@ -142,8 +142,10 @@ export function getAppRoot(
   // else the client root is the same as the server
   root.clientNode = opts.buffer ? createBuffer(root) : root.serverNode;
 
-  // create an overlay that can be used later if a freeze event occurs
-  root.overlay = createOverlay(_document);
+  // create an overlay if not disabled ,that can be used later if a freeze event occurs
+  if (!(opts.disableOverlay || false)) {
```
Comment (mgol) [inline src/lib/api/event.recorder.ts]: `|| false`?

**Comment by ankitu** [comment]
I signed it!

File: src/lib/api/event.recorder.ts (hunk 1 above)
Comment (ankitu) [reply src/lib/api/event.recorder.ts]: Sure, we can omit it here, as we are already setting the default value for `disableOverlay `

Comment (ankitu) [reply src/lib/api/event.recorder.ts]: Done

File: src/lib/api/event.recorder.ts (hunk 2 above)
Comment (ankitu) [reply src/lib/api/event.recorder.ts]: Done

**Comment by ankitu** [comment]
@mgol Feedback addressed, and thanks for the prompt reply

**Comment by ankitu** [comment]
@mgol We are already skipping the overlay creation in `getAppRoot`
```
// create an overlay if not disabled ,that can be used later if a freeze event occurs
  if (!opts.disableOverlay) {
    root.overlay = createOverlay(_document);
  }
```

And in `cleanup`, I guess condition is not required as we are are already checking for `prebootOverlay` id.
```
const prebootOverlay = doc.getElementById('prebootOverlay');
    if (prebootOverlay) {
      prebootOverlay.remove ?
        prebootOverlay.remove() : prebootOverlay.parentNode !== null ?
        prebootOverlay.parentNode.removeChild(prebootOverlay) :
        prebootOverlay.style.display = 'none';
    }
```

File: src/lib/api/event.recorder.ts [hunk 3]
```diff
@@ -262,8 +265,8 @@ export function createListenHandler(
       prebootData.activeNode = undefined;
     }
 
-    // if we are freezing the UI
-    if (eventSelector.freeze) {
+    // if overlay is not disabled and we are freezing the UI
+    if (!opts.disableOverlay && eventSelector.freeze) {
```
[the remark this answers is not in the document]
Comment (ankitu) [reply src/lib/api/event.recorder.ts]: Done

**Pull request merged**